tests/phpunit/elementor/includes/template-library/test-rest-api.php

```diff
@@ -0,0 +1,57 @@
+<?php
+namespace Elementor\Testing\Includes\TemplateLibrary;
+
+use Elementor\Plugin;
+use ElementorEditorTesting\Elementor_Test_Base;
+
+class Test_REST_API extends Elementor_Test_Base {
+	public function test_create_new_template() {
+		$this->act_as_admin();
+
+		$request = new \WP_REST_Request( 'POST', '/wp/v2/elementor-templates' );
+		$request->set_body_params( [
+			'title' => 'My custom template',
+			'document_type' => 'page'
+		] );
+
+		$response = rest_do_request( $request );
+
+		$this->assertEquals( 201, $response->status );
+		$this->assertEquals( 'page', $response->data['document_type'] );
+
+		$document = Plugin::$instance->documents->get( $response->data['id'] );
+
+		$this->assertNotNull( $document );
+		$this->assertEquals( 'page', $document->get_template_type() );
+		$this->assertTrue( $document->is_built_with_elementor() );
+	}
+
+	public function test_create_new_template__unauthorized_creating_template() {
+		$this->act_as_subscriber();
+
+		$request = new \WP_REST_Request( 'POST', '/wp/v2/elementor-templates' );
+		$request->set_body_params( [
+			'title' => 'My custom template',
+			'document_type' => 'page'
+		] );
+
+		$response = rest_do_request( $request );
+
+		$this->assertEquals( 403, $response->status );
+	}
+
+	public function test_create_new_template__document_type_is_required() {
+		$this->act_as_subscriber();
```

Comment:
Why not: 
```suggestion
```

Reply:
Mistake, it should be `act_as_admin` because the test is for the `document_type`